Add stylelint icon

drujbanjo opened the pull request:
# Add New Icon(s)

## Icon Details

| Theme | Icon |
|---------|------|
| auto | ![auto](https://raw.githubusercontent.com/LelouchFR/skill-icons/36b044a32c431c5be537012ce37071b4d0d3f6f9/assets/stylelint-auto.svg) |
| dark | ![dark](https://raw.githubusercontent.com/LelouchFR/skill-icons/36b044a32c431c5be537012ce37071b4d0d3f6f9/assets/stylelint-dark.svg) |
| light | ![light](https://raw.githubusercontent.com/LelouchFR/skill-icons/36b044a32c431c5be537012ce37071b4d0d3f6f9/assets/stylelint-light.svg) |

## Notes

closes  #507

I created an issue, but it wasn't read, so I decided to create the icon myself. This icon was made in Figma. It was taken from the official library source. It has a theme change and is made in accordance with the standards of all skill-icons. The library logo itself changes to black and white depending on the theme.

There were no deviations from the standard style. The background is the same as other icons, radius=60.

black - #131313
white - #F1F1F1

dark bg - #242938
light bg - #F4F2ED

## Checklist

- [X] Added icon(s) to the appropriate directory
- [x] Updated documentation with the new icon(s)
- [X] Verified the icon(s) display correctly
- [X] If the icon has light and dark, verified the auto switch works correctly
- [X] The Svg does not contain any sort of strokes

drujbanjo committed:
added stylelint icon

drujbanjo committed:
added stylelint icons

LelouchFR reviewed (changes requested):
for the light and dark icons, please use `#F1F1F1` instead of the `white` and `#131313` for the `black`.

please, for the auto icon, follow the same code structure as other icons.

add `stylelint` id to the svg tag.

for the styles, use this:

```css
#stylelint {
    rect {fill: #242938}
    path[fill="#F1F1F1"] {fill: #F1F1F1}

    @media (prefers-color-scheme: light) {
        rect {fill: #F4F2ED}
        path[fill="#F1F1F1"] {fill: #131313}
    }
}
```

which will give for the auto icon:

```svg
<svg width="256" height="256" viewBox="0 0 256 256" fill="none" xmlns="http://www.w3.org/2000/svg" id="stylelint">
<style>
#stylelint {
    rect {fill: #242938}
    path[fill="#F1F1F1"] {fill: #F1F1F1}

    @media (prefers-color-scheme: light) {
        rect {fill: #F4F2ED}
        path[fill="#F1F1F1"] {fill: #131313}
    }
}
</style>
<rect width="256" height="256" rx="60" fill="#242938"/>
<path d="M129.227 217.711C148.94 147.221 148.368 146.903 162.617 96.6418L166.044 98.6206C167.905 99.6959 169.429 98.8166 169.429 96.6715V72.7026C173.21 59.4595 176.174 49.2264 177.042 46.1494H192.479L218.125 69.9074L201.282 84.4631L214.018 98.2325L130.079 217.711C129.669 218.386 129.097 218.232 129.227 217.711Z" fill="#F1F1F1"/>
<path d="M138.398 65.777C139.155 65.777 139.774 66.1961 139.774 66.7163V79C139.774 79.5189 139.155 79.9406 138.398 79.9406H117.662C116.9 79.9406 116.282 79.5206 116.282 79V66.7163C116.282 66.1979 116.899 65.777 117.662 65.777H138.398ZM163.991 52.176L141.18 65.3447C141.444 65.7472 141.605 66.211 141.605 66.7167V79.0009C141.605 79.6383 141.359 80.2127 140.951 80.6765L163.991 93.9774C165.562 94.8843 166.85 94.1423 166.85 92.3306V53.8245C166.85 52.0111 165.562 51.2712 163.991 52.176ZM114.451 79V66.7163C114.451 66.2132 114.612 65.7494 114.873 65.3469L92.062 52.176C90.4923 51.2712 89.2051 52.0115 89.2051 53.8245V92.3302C89.2051 94.1414 90.4905 94.8852 92.062 93.9765L115.101 80.6743C114.696 80.2105 114.451 79.6374 114.451 79Z" fill="#F1F1F1"/>
<path d="M125.921 217.711H126.773C107.063 147.221 107.631 146.903 93.3828 96.6418L89.9559 98.6206C88.0952 99.6959 86.5709 98.8166 86.5709 96.6715V72.7026C82.7883 59.4595 79.8264 49.2264 78.9563 46.1494H63.5199L37.875 69.9074L54.7161 84.4631L41.9805 98.2325L125.921 217.711Z" fill="#F1F1F1"/>
<path d="M127.375 105.586C130.814 105.586 133.601 102.799 133.601 99.3604C133.601 95.9218 130.814 93.1343 127.375 93.1343C123.937 93.1343 121.149 95.9218 121.149 99.3604C121.149 102.799 123.937 105.586 127.375 105.586Z" fill="#F1F1F1"/>
<path d="M127.375 142.607C130.814 142.607 133.601 139.819 133.601 136.381C133.601 132.942 130.814 130.155 127.375 130.155C123.937 130.155 121.149 132.942 121.149 136.381C121.149 139.819 123.937 142.607 127.375 142.607Z" fill="#F1F1F1"/>
<path d="M127.375 179.632C130.814 179.632 133.601 176.844 133.601 173.406C133.601 169.968 130.814 167.18 127.375 167.18C123.937 167.18 121.15 169.968 121.15 173.406C121.15 176.844 123.937 179.632 127.375 179.632Z" fill="#F1F1F1"/>
</svg>
```

drujbanjo committed:
correct icons

drujbanjo commented:
> for the light and dark icons, please use `#F1F1F1` instead of the `white` and `#131313` for the `black`.
> 
> please, for the auto icon, follow the same code structure as other icons.
> 
> add `stylelint` id to the svg tag.
> 
> for the styles, use this:
> 
> ```css
> #stylelint {
>     rect {fill: #242938}
>     path[fill="#F1F1F1"] {fill: #F1F1F1}
> 
>     @media (prefers-color-scheme: light) {
>         rect {fill: #F4F2ED}
>         path[fill="#F1F1F1"] {fill: #131313}
>     }
> }
> ```
> 
> which will give for the auto icon:
> 
> ```
> <svg width="256" height="256" viewBox="0 0 256 256" fill="none" xmlns="http://www.w3.org/2000/svg" id="stylelint">
> <style>
> #stylelint {
>     rect {fill: #242938}
>     path[fill="#F1F1F1"] {fill: #F1F1F1}
> 
>     @media (prefers-color-scheme: light) {
>         rect {fill: #F4F2ED}
>         path[fill="#F1F1F1"] {fill: #131313}
>     }
> }
> </style>
> <rect width="256" height="256" rx="60" fill="#242938"/>
> <path d="M129.227 217.711C148.94 147.221 148.368 146.903 162.617 96.6418L166.044 98.6206C167.905 99.6959 169.429 98.8166 169.429 96.6715V72.7026C173.21 59.4595 176.174 49.2264 177.042 46.1494H192.479L218.125 69.9074L201.282 84.4631L214.018 98.2325L130.079 217.711C129.669 218.386 129.097 218.232 129.227 217.711Z" fill="#F1F1F1"/>
> <path d="M138.398 65.777C139.155 65.777 139.774 66.1961 139.774 66.7163V79C139.774 79.5189 139.155 79.9406 138.398 79.9406H117.662C116.9 79.9406 116.282 79.5206 116.282 79V66.7163C116.282 66.1979 116.899 65.777 117.662 65.777H138.398ZM163.991 52.176L141.18 65.3447C141.444 65.7472 141.605 66.211 141.605 66.7167V79.0009C141.605 79.6383 141.359 80.2127 140.951 80.6765L163.991 93.9774C165.562 94.8843 166.85 94.1423 166.85 92.3306V53.8245C166.85 52.0111 165.562 51.2712 163.991 52.176ZM114.451 79V66.7163C114.451 66.2132 114.612 65.7494 114.873 65.3469L92.062 52.176C90.4923 51.2712 89.2051 52.0115 89.2051 53.8245V92.3302C89.2051 94.1414 90.4905 94.8852 92.062 93.9765L115.101 80.6743C114.696 80.2105 114.451 79.6374 114.451 79Z" fill="#F1F1F1"/>
> <path d="M125.921 217.711H126.773C107.063 147.221 107.631 146.903 93.3828 96.6418L89.9559 98.6206C88.0952 99.6959 86.5709 98.8166 86.5709 96.6715V72.7026C82.7883 59.4595 79.8264 49.2264 78.9563 46.1494H63.5199L37.875 69.9074L54.7161 84.4631L41.9805 98.2325L125.921 217.711Z" fill="#F1F1F1"/>
> <path d="M127.375 105.586C130.814 105.586 133.601 102.799 133.601 99.3604C133.601 95.9218 130.814 93.1343 127.375 93.1343C123.937 93.1343 121.149 95.9218 121.149 99.3604C121.149 102.799 123.937 105.586 127.375 105.586Z" fill="#F1F1F1"/>
> <path d="M127.375 142.607C130.814 142.607 133.601 139.819 133.601 136.381C133.601 132.942 130.814 130.155 127.375 130.155C123.937 130.155 121.149 132.942 121.149 136.381C121.149 139.819 123.937 142.607 127.375 142.607Z" fill="#F1F1F1"/>
> <path d="M127.375 179.632C130.814 179.632 133.601 176.844 133.601 173.406C133.601 169.968 130.814 167.18 127.375 167.18C123.937 167.18 121.15 169.968 121.15 173.406C121.15 176.844 123.937 179.632 127.375 179.632Z" fill="#F1F1F1"/>
> </svg>
> ```

I added #F1F1F1 and #131313, added #stylelint. Theme switching works correctly.

drujbanjo committed:
update documentation

drujbanjo committed:
update documentation

drujbanjo committed:
reupdate documentation

drujbanjo committed:
reupdate documentation

drujbanjo commented:
I don't know how update documentation because It is very big table of icons. I tried to do it, but I couldn't. I can create icons but don't update documentation

LelouchFR commented:
> I don't know how update documentation because It is very big table of icons. I tried to do it, but I couldn't. I can create icons but don't update documentation

to update the documentation, please use the script `./.github/readme-format.sh`

drujbanjo committed:
update documentation

drujbanjo commented:
> > I don't know how update documentation because It is very big table of icons. I tried to do it, but I couldn't. I can create icons but don't update documentation
> 
> to update the documentation, please use the script `./.github/readme-format.sh`

Oh thank. I updated documentation with that command

LelouchFR reviewed (approved):
LGTM
